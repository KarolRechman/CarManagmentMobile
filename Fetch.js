import React, { useState, useEffect } from "react";
import axios from "axios";
import { Button, Text, TextInput, View } from 'react-native'
import { act } from 'react-dom/test-utils';

const useAxios = (url, setData) => {
    useEffect(
        () => {
            let mounted = true;

            const loadData = async () => {
                const result = await axios.get(url);
                if (mounted) {
                    act(() => {
                        setData(result.data);
                    });
                }
            };
            loadData();

            return () => {
                mounted = false;
            };
        },
        [url]
    );
};

export default function Fetch({ url }) {
    const [data, setData] = useState(null);
    useAxios(url, setData);

    if (!data) {
        return (<View><Text testID="loading">Loading data...</Text></View>);
    } else {
        return (<View><Text testID="resolved">{data.model}</Text></View>);
    }
}