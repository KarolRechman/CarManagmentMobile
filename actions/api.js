import axios from "axios";

const mainURL = "http://192.168.56.1:7000/"

export const API_TYPES = {
    USER: `${mainURL}User`,
    CAR: `${mainURL}Car`,
    SPENDINGS: `${mainURL}Spendings`,
    COSTS: `${mainURL}Costs`,
    TRANSACTIONS: `${mainURL}Transactions`,
}

/**
 * test3@gamil.com
 * 1234Aa
 */
export default {

    request(url) {
        return {
            fetchAll: () => axios.get(url),
            fetchById: id => axios.get(url + id),
            create: newRecord => axios.post(url,newRecord),
            update: (id, updatedRecord) => axios.put(url + id, updatedRecord),
            delete: id => axios.delete(url + id),
            userLogin: user => axios.post(url + "/login", user),
            userRegister: (password, user) => axios.post(url + `/register?password=${password}`, user),
            fetchUserCars: id => axios.get(url + "/GetUserCars" + id),
            fetchLogs: id => axios.get(url + "/GetLogs" + id),
            fetchSpendings: id => axios.get(url + "/GetSpendings" + id),
            sendEmail: email => axios.post(url + "/Email", email)
        }
    }
}


