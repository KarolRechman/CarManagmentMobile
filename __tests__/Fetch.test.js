import React from "react";
import { render, cleanup, waitFor } from '@testing-library/react-native'
import "@testing-library/jest-dom/extend-expect";
import axiosMock from "axios";
import Fetch from "../Fetch";
import api, { API_TYPES } from "../actions/api";


afterEach(cleanup);
jest.setTimeout(30000);

test("fetches and displays data", async () => {
    // We'll be explicit about what data Axios is to return when `get` is called.
    axiosMock.get.mockResolvedValueOnce({ data: { model: "Astra" } });
    const url = "/greeting";
    const { getByTestId, getByText, queryByTestId, toJSON } = render(<Fetch url={url} />);

    // const input = getByTestId('resolved')
    // console.log(input)

    expect(getByTestId("loading").props.children).toBe("Loading data...");



     await waitFor(() => expect(getByTestId('resolved')).toBeTruthy())


    // expect(getByTestId('resolved').props.children).toBe("Astra")
    expect(toJSON()).toMatchSnapshot()

    // expect(axiosMock.get).toHaveBeenCalledTimes(1);
    // expect(axiosMock.get).toHaveBeenCalledWith(url);
});



