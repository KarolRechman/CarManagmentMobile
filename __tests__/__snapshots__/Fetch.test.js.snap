// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches and displays data 1`] = `
<View>
  <Text
    testID="resolved"
  >
    Astra
  </Text>
</View>
`;
